pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
The primary reason the agent is unlikely to ever stay in this sleep for the full 24 hours is because a reboot has just been requested. Its kinda unclear what we ever gained by having this sleep, its not like the agent tried again after 24 hours - it just stopped running.